cfgov/ask_cfpb/views.py

```diff
@@ -109,37 +111,100 @@ def ask_search(request, language='en', as_json=False):
                     'text': result.text,
                     'preview': result.preview,
                 }
-                for result in search.queryset
+                for result in search.get('results', [])
             ]
         }
-        json_results = json.dumps(results)
+        json_results = json.dumps(payload)
         return HttpResponse(json_results, content_type='application/json')
 
     results_page.query = search_term
     results_page.result_query = search.search_term
     results_page.suggestion = search.suggestion
     results_page.answers = [
         (result.url, result.autocomplete, result.preview)
-        for result in search.queryset
+        for result in search.get("results")
     ]
     return results_page.serve(request)
 
 
-def ask_autocomplete(request, language='en'):
-    term = request.GET.get(
-        'term', '').strip().replace('<', '')
-    if not term:
-        return JsonResponse([], safe=False)
+def ask_search_es7(request, language='en', as_json=False):
+    if 'selected_facets' in request.GET:
+        return redirect_ask_search(request, language=language)
+    language_map = {
+        'en': 'ask-cfpb-search-results',
+        'es': 'respuestas'
+    }
+    results_page = get_object_or_404(
+        AnswerResultsPage,
+        language=language,
+        slug=language_map[language]
+    )
+
+    # If there's no query string, don't search
+    search_term = request.GET.get('q', '')
+    # Check if we want to use the suggestion or not
+    suggest = request.GET.get('correct', '1') == '1'
+    if not search_term:
+        results_page.query = ''
+        results_page.result_query = ''
+        return results_page.serve(request)
+    search_obj = AnswerPageSearch(search_term, language=language)
+    response = search_obj.search()
+    if not response.get('results') and suggest:
+        response = search_obj.suggest()
+    if as_json:
+        result_list = [{
+            'question': result.autocomplete,
+            'url': result.url,
+            'text': result.text,
+            'preview': result.preview}
+            for result in response.get('results')
+        ]
+        payload = {
+            'query': search_term,
+            'result_query': make_safe(search_term).strip(),
+            'suggestion': search_obj.suggest().get("suggestion"),
+            'results': result_list
+        }
+        json_results = json.dumps(payload)
+        return HttpResponse(json_results, content_type='application/json')
+    results_page.query = search_term
+    results_page.result_query = response.get('search_term')
+    results_page.suggestion = response.get('suggestion')
+    results_page.answers = [
+        (result.url, result.autocomplete, result.preview)
+        for result in response['results']
+    ]
+    return results_page.serve(request)
+
 
+# def ask_autocomplete(request, language='en'):
```

Comment:
This is the ES 2.4 implementation, should remain functional and trigger the es7 version when a flag is checked.

Reply:
Done